addons/pinniped/config-controller/hack/run.sh

```diff
@@ -5,22 +5,141 @@
+
+# location to deploy the image
+REGISTRY_NAME="harbor-repo.vmware.com"
+REGISTRY_PROJECT="tkgiam"
+REGISTRY="${REGISTRY_NAME}/${REGISTRY_PROJECT}"
+
+# name of image to match that defined in /packages/management/tanzu-auth
+CONTROLLER_MANAGER_NAME="${APP_NAME}-controller-manager"
+CONTROLLER_MANAGER_PACKAGE_NAME="${CONTROLLER_MANAGER_NAME}-package"
+CONTROLLER_NAMESPACE_NAME="${APP_NAME}"
 
 # Always run from config-controller directory for reproducibility
 cd "${MY_DIR}/.."
 
-tag="dev"
-# tag="$(uuidgen)" # Uncomment to create random image every time
-image="harbor-repo.vmware.com/tkgiam/$(whoami)/pinniped-config-controller-manager:$tag"
+TAG="dev"
+# TAG="$(uuidgen)" # Uncomment to create random image every time
+
+# build the tanzu-auth-controller-manager image
+# --------------------------------------------------------
+CONTROLLER_IMAGE="${REGISTRY}/$(whoami)/${CONTROLLER_MANAGER_NAME}:${TAG}"
+echo -e "${YELLOW}building ${CONTROLLER_MANAGER_NAME} image and pushing to ${CONTROLLER_IMAGE}...${DEFAULT}"
+
+# push the tanzu-auth-controller-manager to the registry
+docker build -t "${CONTROLLER_IMAGE}" .
+docker push "${CONTROLLER_IMAGE}"
+
+
+# build the tanzu-auth-controller-manager-package image
+# --------------------------------------------------------
+PACKAGE_IMAGE="${REGISTRY}/$(whoami)/${CONTROLLER_MANAGER_PACKAGE_NAME}:${TAG}"
+echo -e "${YELLOW}building ${CONTROLLER_MANAGER_PACKAGE_NAME} image and pushing to ${PACKAGE_IMAGE}...${DEFAULT}"
 
+# generate new RBAC into /packages/management/tanzu-auth/bundle/config
+echo -e "${YELLOW}generating RBAC...${DEFAULT}"
 ./hack/generate.sh
-kubectl apply -f "${RBAC_FILE}"
 
-docker build -t "$image" .
-docker push "$image"
-ytt --data-value "image=$image" -f "${DEPLOYMENT_FILE}" | kbld -f - | kapp deploy -a pinniped-config-controller-manager -f - -y
-kubectl logs -n pinniped deploy/pinniped-config-controller-manager -f
+# the namespace must exist before the package can be deployed
+echo -e "${YELLOW}creating namespace ${APP_NAME}...${DEFAULT}"
+ytt --data-value "tanzuAuth.namespace=${APP_NAME}" --file "${TANZU_AUTH_CONFIG_DIR}/namespace.yaml" | kubectl apply -f -
+
+echo -e "${YELLOW}rebuilding image lock file with ytt and kbld...${DEFAULT}"
+ytt --data-value "image=${CONTROLLER_IMAGE}" --file "${TANZU_AUTH_DEFAULT_VALUES_FILE}" --file "${TANZU_AUTH_CONFIG_DIR}" \
+  | kbld --file - --imgpkg-lock-output "${TANZU_AUTH_BUNDLE_DIR}/.imgpkg/images.yml"
+
+# push the tanzu-auth-controller-manager-package to the registry
+echo -e "${YELLOW}pushing ${PACKAGE_IMAGE} to registry with imgpkg...${DEFAULT}"
+imgpkg push --bundle "${PACKAGE_IMAGE}" --file "${TANZU_AUTH_BUNDLE_DIR}"
+
+# inject the package name into the package file & deploy
+echo -e "${YELLOW}creating package ${TANZU_AUTH_PACKAGE_FILE} with image ${PACKAGE_IMAGE}...${DEFAULT}"
+yq e ".spec.template.spec.fetch[0].imgpkgBundle.image = \"${PACKAGE_IMAGE}\"" "${TANZU_AUTH_PACKAGE_FILE}" \
```

Comment:
For reference, this will generate a package yaml something like:

```yaml
# package
apiVersion: data.packaging.carvel.dev/v1alpha1
kind: Package
metadata:
  annotations:
  name: tanzu-auth.tanzu.vmware.com.1.6.0
  namespace: tanzu-auth
spec:
  licenses:
  - Apache 2.0
  refName: tanzu-auth.tanzu.vmware.com
  releasedAt: null
  template:
    spec:
      deploy:
      - kapp: {}
      fetch:
      - imgpkgBundle:
          image: harbor-repo.vmware.com/tkgiam/ben/tanzu-auth-controller-manager-package:dev
      template:
      - ytt:
          paths:
          - config/
      - kbld:
          paths:
          - '-'
          - .imgpkg/images.yml
  valuesSchema:
    openAPIv3: null
  version: 1.6.0
```